SeQuant/core/eval_expr.cpp

```diff
@@ -258,48 +258,45 @@ std::pair<container::svector<Index>,  // bra
           container::svector<Index>   // ket
           >
 target_braket(Tensor const& t1, Tensor const& t2) noexcept {
-  using ranges::views::keys;
-  using ranges::views::values;
-  using ranges::views::zip;
-  using index_container = container::svector<Index>;
-
-  auto remove_item = [](auto& vec, size_t pos) -> void {
-    std::swap(vec[pos], vec.back());
-    vec.pop_back();
-  };
-
-  auto left = zip(t1.bra(), t1.ket()) | ranges::to_vector;
-  auto right = zip(t2.bra(), t2.ket()) | ranges::to_vector;
-
-  while (!right.empty()) {
-    for (std::size_t rr = 0; rr < right.size(); ++rr) {
-      auto& [rb, rk] = right[rr];
-      for (std::size_t ll = 0; ll < left.size(); ++ll) {
-        auto& [lb, lk] = left[ll];
-        if (lb == rk && rb == lk) {
-          remove_item(left, ll);
-          remove_item(right, rr);
-          goto next_contract;
-        } else if (lb == rk) {
-          std::swap(lk, rk);
-          remove_item(left, ll);
-          goto next_contract;
-        } else if (lk == rb) {
-          std::swap(lb, rb);
-          remove_item(left, ll);
-          goto next_contract;
-        }
-      }  // ll
-      left.emplace_back(rb, rk);
-      remove_item(right, rr);
-    }  // rr
-  next_contract:
-      /* just point to the start of the while loop */;
-  }
-  // the result is now in left
-
-  return {keys(left) | ranges::to<index_container>,
-          values(left) | ranges::to<index_container>};
+  using ranges::contains;
+  using ranges::views::concat;
+  using ranges::views::filter;
+  using idx_container = container::svector<Index>;
+
+  const auto left_bra = t1.bra() | ranges::to_vector;
```

Comment:
any reason to convert these to vectors, instead of using range `t1.bra()` directly in place of `left_bra`?

Reply:
Fixed in https://github.com/ValeevGroup/SeQuant/commit/21328826ce6afc227118b3197324b672e7c32ca7